logos-codegen/src/mir.rs

```diff
@@ -69,7 +69,12 @@ impl Mir {
             Mir::Concat(concat) => concat.iter().map(Mir::priority).sum(),
             Mir::Alternation(alt) => alt.iter().map(Mir::priority).min().unwrap_or(0),
             Mir::Class(_) => 1,
-            Mir::Literal(_) => 2,
+            Mir::Literal(lit) => {
+                match std::str::from_utf8(&lit.0) {
+                    Ok(s) => 2 * s.len(),
+                    Err(_) => 2 * lit.0.len(),
+                }
```

Comment:
I guess you're doing this because of unicode strings?
Is it possible to mix string and byte pattern in logos?
If so, can you provide a testcase where this is covered?

If not, I think doing the simple solution, e.g. `2 * lit.0.len()` is enough here. But I don't have the last word of course ;)

Reply:
Yes, I should come with a test for that :-)

But I did that because `Mir::Literal` can be constructed either from a `char` or from an `u8`. But a single `char`, which should have priority `2`, can be made of up to 4 bytes. So using `2 * lit.0.len()` for such a `char` could make the priority much higher. I will try to write some tests for that :-)

Reply:
Ohh I made the dummy mistake, but `str`'s length should be `s.chars().count()` :'-)

Reply:
But then we talk about grapheme clusters? I mean `ẽ` could either be just `ẽ` or `e~`.
Please take a look at: https://stackoverflow.com/questions/58770462/how-to-iterate-over-unicode-grapheme-clusters-in-rust

Reply:
We talk about unicode chars, see [regex#54](https://github.com/rust-lang/regex/issues/54).

I don't think `regex-syntax` or `logos` supports grapheme clusters, so I don't see why we should support them now?